tools/autograd/gen_annotated_fn_args.py

```diff
@@ -13,65 +13,69 @@
 torch/testing/_internal/generated
 """
 
-from .utils import write, CodeTemplate
-from .gen_python_functions import (
-    get_py_nn_functions,
-    get_py_torch_functions,
-    get_py_variable_methods,
-    op_name,
-)
+from collections import defaultdict
 import argparse
+import os
 import textwrap
-from .gen_autograd import load_aten_declarations
 
-
-def gen_annotated(aten_path, out, template_path):
-    declarations = load_aten_declarations(aten_path)
-    annotated_args = []
-    for func in recurse_dict(get_py_torch_functions(declarations)):
-        annotated_args.append(process_func("torch._C._VariableFunctions", func))
-
-    for func in recurse_dict(get_py_nn_functions(declarations)):
-        annotated_args.append(process_func("torch._C._nn", func))
-
-    for func in recurse_dict(get_py_variable_methods(declarations)):
-        annotated_args.append(process_func("torch.Tensor", func))
-
-    annotated_args = textwrap.indent("\n".join(annotated_args), "    ")
-    env = {"annotated_args": annotated_args}
-    PY_ANNOTATED_ARGS = CodeTemplate.from_file(template_path + '/templates/annotated_fn_args.py')
-    write(out, 'annotated_fn_args.py', PY_ANNOTATED_ARGS, env)
-
-
-def process_func(namespace, func):
-    args = func["arguments"]
-    out_args = []
-    for arg in args:
-        if 'default' in arg or arg.get('kwarg_only', False) or arg.get('output', False):
+from typing import Dict, List, Any
+
+from tools.codegen.gen import with_native_function, parse_native_yaml, FileManager
+from tools.codegen.model import *
+import tools.codegen.api.python as python
+from .gen_python_functions import should_generate_py_binding, is_py_torch_function, is_py_nn_function, is_py_variable_method
+
+def gen_annotated(native_yaml_path: str, out: str, autograd_dir: str) -> None:
+    native_functions = parse_native_yaml(native_yaml_path)
+    mappings = (
+        (is_py_torch_function, 'torch._C._VariableFunctions'),
+        (is_py_nn_function, 'torch._C._nn'),
+        (is_py_variable_method, 'torch.Tensor'),
+    )
+    annotated_args: List[str] = []
+    for pred, namespace in mappings:
+        groups: Dict[str, List[NativeFunction]] = defaultdict(list)
+        for f in native_functions:
+            if not should_generate_py_binding(f) or not pred(f):
+                continue
+            groups[str(f.func.name.name)].append(f)
+        for group in groups.values():
+            for f in group:
+                annotated_args.append(f'{namespace}.{process_func(f)}')
+
+    template_path = os.path.join(autograd_dir, 'templates')
+    fm = FileManager(install_dir=out, template_dir=template_path, dry_run=False)
+    fm.write_with_template('annotated_fn_args.py', 'annotated_fn_args.py', lambda: {
```

Comment:
...so per our thread in #47818, is this gonna say "generated by aten/src/ATen/gen.py"? 😁  

oh also, curious how changes in these comment lines play in the byte-for-byte comparisons